playground/hmr/__tests__/hmr.spec.ts

```diff
@@ -780,6 +780,57 @@ if (import.meta.hot) {
     )
   })
 
+  test('delete file should not brake hmr', async () => {
+    await page.goto(viteTestUrl)
+
+    await untilUpdated(
+      () => page.textContent('.intermediate-file-delete-display'),
+      'count is 1',
+    )
+
+    // add state
+    await page.click('.intermediate-file-delete-increment')
+    await untilUpdated(
+      () => page.textContent('.intermediate-file-delete-display'),
+      'count is 2',
+    )
+
+    // update import, hmr works
+    editFile('intermediate-file-delete/index.js', (code) =>
+      code.replace("from './re-export.js'", "from './display.js'"),
+    )
+    editFile('intermediate-file-delete/display.js', (code) =>
+      code.replace('count is ${count}', 'count is ${count}!'),
+    )
+    await untilUpdated(
+      () => page.textContent('.intermediate-file-delete-display'),
+      'count is 2!',
+    )
+
+    // remove unused file, page reload because it's considered entry point now
```

Comment:
would it be possible to avoid page reload or hmr when removing `re-export.js`, ideally ` re-export.js` should be removed from `moduleGraph` when replacing `from './re-export.js'` with `from './display.js'`

Reply:
After investigation it seems unsafe to remove the entry when it has no importer left without knowing if the node was present before because it's referenced from the update module or maybe also from an html file.